src/tests/JIT/HardwareIntrinsics/X86/Shared/_BinaryOpTestTemplate.template

```diff
@@ -179,6 +188,57 @@ namespace JIT.HardwareIntrinsics.X86
             ValidateResult(_dataTable.inArray1Ptr, _dataTable.inArray2Ptr, _dataTable.outArrayPtr);
         }
 
+        public void RunBroadcastScenario()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBroadcastScenario));
+
+            var result = {Isa}.{Method}(
+                Unsafe.Read<{Op1VectorType}<{Op1BaseType}>>(_dataTable.inArray1Ptr),
+                {Op2VectorType}.Create<{Op2BaseType}>(Unsafe.Read<{Op2BaseType}>(_dataTable.inArray2Ptr))
+            );
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateResult(_dataTable.inArray1Ptr, _dataTable.inArray2Ptr, _dataTable.outArrayPtr, isBroadcast: true);
+        }
+
+        public void RunMaskingScenario()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunMaskingScenario));
+
+            var op1 = Unsafe.Read<{Op1VectorType}<{Op1BaseType}>>(_dataTable.inArray1Ptr);
+
+            var result = {RetVectorType}.ConditionalSelect(
```

Comment:
Can you also have a test where for `ConditionalSelect` you pass a non-zero value and verify it works functionally correct?

Reply:
I can add another test, yes. Was trying to keep this a bit simpler to start since handling having more complex CndSel tests for all intrinsics can be tricky.

Reply:
i found some issues when I added those, specifically when I pass `op1` and `op2` as the 3rd parameter of `CndSel`.

Reply:
I’ll get them added later tonight, but worth noting that it’s less likely to be problematic for x64 due to the variable width encoding and most instructions being non-RMW. That is, there is typically no special consideration like needing to pick between `movprfx; add (predicated)` vs `add (predicated)` vs `add; sel`

Reply:
Added